RocketChat/Rocket.Chat.ReactNative
# [FIX] App not prompting join code for password protected channels

Co-authored-by: Vitor Leal <vitor_leal2201@hotmail.com>
Co-authored-by: Fernando Aguilar <fernando.aguilar@hotmail.com.br>



## Proposed changes


Creating an input for password insertion when the channel is password protected, and creating a status in RoomView to store the joinCode attribute described in the documentation (https://docs.rocket.chat/api/rest-api/methods/channels/join) to send in the post when the channel is password protected.

## Issue(s)	

Closes #2258 

## How to test or reproduce


To reproduce, just try to enter a password-protected channel, no field will appear to enter the password, and it will not be possible to enter.

## Screenshots

![](https://media.giphy.com/media/7a3zYVjxqTM0CrhGgF/giphy.gif)
![Screen Shot 2020-10-13 at 1 57 13 PM](https://user-images.githubusercontent.com/29778115/95894056-f1380780-0d5e-11eb-8c33-29dc6cd865e3.png)
![Screen Shot 2020-10-13 at 1 57 18 PM](https://user-images.githubusercontent.com/29778115/95894059-f301cb00-0d5e-11eb-9caa-67d8280a217a.png)
![Screen Shot 2020-10-13 at 2 11 09 PM](https://user-images.githubusercontent.com/29778115/95894064-f432f800-0d5e-11eb-9e58-a2d8e3f0b3d4.png)


## Types of changes




## Checklist



## Further comments
